### PR TITLE
Added eslint and prettier

### LINKED ISSUE #13: Add EsLint To The Repository
**Is your feature request related to a problem? Please describe.**
In order to keep code tidy in the repository and save time when reviewing indentation, semicolons, etc... in code review, it is proposed to add EsLint and StyleLint to the repository.

**Describe the solution you'd like**
Install EsLint and StyleLint to the repository.

**Additional context**
I tried using:
```
npx eslint . && npx stylelint .'
```

And they guided me step by step how to install it, configuring it was easy and then running it to fix the code.

### PR DESCRIPTION
**Pull request by e-azocar**
### Description

Added `eslint` as Javascript linter, `stylelint` as CSS linter  and `prettier` to manage the code format

fixes #13

### CODE REVIEW

File: src/js/utils.js [hunk 1]
```diff
@@ -1,66 +1,69 @@
...
-    user,
-    repository,
-    typeOfSearch,
-    condition,
-    date
-  } = formData;
+  const { user, repository, typeOfSearch, condition, date } = formData;
 
-  let searchCondition = '';
+  let searchCondition = "";
```
Comment (heliomar-pena) [inline src/js/utils.js]: Can you add a rule to EsLint to change double quotes for single quotes in javascript code?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+		"doc": "docs"
+	},
+	"scripts": {
+		"lint": "eslint --ext .js src",
+		"lint:fix": "eslint src --ext .js --fix",
+		"format": "prettier --write \"**/*.+(js|html|css|md)\""
+	},
+	"repository": {
+		"type": "git",
+		"url": "git+https://github.com/Azordev/search-by-time-github.git"
```
Comment (heliomar-pena) [inline package.json:16]: this `git+` url is ok? 

I mean, it really start for: "git+"?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	},
+	"keywords": [],
+	"author": "",
+	"license": "ISC",
+	"bugs": {
+		"url": "https://github.com/Azordev/search-by-time-github/issues"
+	},
+	"homepage": "https://github.com/Azordev/search-by-time-github#readme",
+	"devDependencies": {
+		"eslint": "^8.11.0"
```
Comment (heliomar-pena) [inline package.json]: Please freeze dependency

**Review by heliomar-pena — APPROVED** [review]
Good work! I have left you some comments

File: package.json (hunk 2 above)
Comment (Israel-Laguan) [reply package.json:16]: yes, that is the correct format I think

File: index.html [hunk 4]
```diff
@@ -1,191 +1,191 @@
...
+								<button id="goback" class="btn--back p-absolute mt-6">
+									<img
+										src="./src/assets/arrow.png"
+										style="margin-top: 3px"
+										width="15px"
+										height="15px"
+										alt=""
+									/>
+								</button>
+							</div>
```
Comment (heliomar-pena) [inline index.html]: It's using tabs instead of spaces?

File: src/css/style.css [hunk 5]
```diff
@@ -1,151 +1,151 @@
...
-  --radius: 10px;
+	--gray: #c4c4c4;
+	--black: #000000;
+	--white: #ffffff;
+	--gray-op-1: #c9d1d9;
+	--button-dark-blue: #001e66;
+	--button-light-blue: #76a4ff;
+	--input-border-blue: #3878f9;
+	--gray-op-2: rgb(139, 148, 158, 0.2);
+	--radius: 10px;
```
Comment (heliomar-pena) [inline src/css/style.css]: It's using tabs instead of spaces?

**Review by heliomar-pena — COMMENTED** [review]
Please check comments. Good work with the changes

File: src/css/style.css (hunk 5 above)
Comment (e-azocar) [reply src/css/style.css]: Currently is using tabs, but, I'll change it to spaces

File: .stylelintrc.json [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{
+  "rules": {
+    "color-no-invalid-hex": true
+  }
+}
```
Comment (Israel-Laguan) [inline .stylelintrc.json]: Please add new line to avoid EOF issue

File: src/css/style.css [hunk 7]
```diff
@@ -1,151 +1,151 @@
...
+	color: var(--black);
+	border-radius: var(--radius);
+	border: 1px solid var(--gray);
+	background-color: var(--white);
+	font-family: Montserrat, sans-serif;
 }
 .input:focus {
-  outline-width: 0px;
-  border-color: var(--input-border-blue);
+	outline-width: 0px;
```
Comment (Israel-Laguan) [inline src/css/style.css]: px is not needed when 0

File: src/css/style.css [hunk 8]
```diff
@@ -1,151 +1,151 @@
...
+	display: grid;
+	padding: 30px;
+	max-width: 450px;
+	max-height: 388px;
+	min-width: min-content;
+	width: 90%;
+	margin: auto;
+	border-radius: 13px;
+	box-sizing: border-box;
+	grid-template-columns: 1fr;
```
Comment (Israel-Laguan) [inline src/css/style.css]: If is only one column, why not use flex?

File: src/css/style.css [hunk 9]
```diff
@@ -1,151 +1,151 @@
...
+	border: 2px solid var(--gray);
+	background-color: var(--white);
+	box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
```
Comment (Israel-Laguan) [inline src/css/style.css]: px is not needed when 0

File: src/css/style.css [hunk 10]
```diff
@@ -1,151 +1,151 @@
...
 }
 .calendary_filter__date_search:focus {
-  outline-width: 0px;
-  border-color: var(--input-border-blue);
+	outline-width: 0px;
```
Comment (Israel-Laguan) [inline src/css/style.css]: px is not needed when 0

File: src/css/style.css [hunk 11]
```diff
@@ -1,151 +1,151 @@
...
-  -webkit-box-sizing: border-box;
-  -moz-box-sizing: border-box;
-  box-sizing: border-box;
+	margin: 0;
+	padding: 0;
+	-webkit-box-sizing: border-box;
+	-moz-box-sizing: border-box;
+	box-sizing: border-box;
 }
 :root {
```
Comment (Israel-Laguan) [inline src/css/style.css:8]: Please add default font here

**Review by Israel-Laguan — APPROVED** [review]

File: src/js/utils.js [hunk 12]
```diff
@@ -1,112 +1,24 @@
...
-  const searchDate = searchCondition && date ? `closed:${searchCondition}${date}` : undefined;
-
-  return {
-    user,
-    repository,
-    typeOfSearch: searchTypeOfSearch,
-    date: searchDate
-  };
+.d-inline {
+  display: inline;
```
Comment (Israel-Laguan) [inline src/js/utils.js]: Please check

**Review by heliomar-pena — APPROVED** [review]
Good work!

**Pull request merged**